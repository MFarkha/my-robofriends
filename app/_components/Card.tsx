import React from "react";
import { IRobot } from "../page";

type CardProps = {
  robot: IRobot;
  key: number;
};

const Card = ({ robot }: CardProps) => {
  const { name, email, id } = robot;
  return (
    <div className="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5">
      <img
        alt="robots"
        src={`https://robohash.org/${id}.png?size=200x200`}
        width="200"
        height="200"
      ></img>
      <div>
        <h2 className="f5">{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
};

export default Card;
