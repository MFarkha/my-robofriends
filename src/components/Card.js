import React from 'react'

const Card = ({name, email, id}) => {
    return (
        <div className='tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5'>
            <img alt='robots' src={`https://robohash.org/${id}.png?size=200x200`} width='200' height='200'></img>
            <div>
            <h2 className='f5'>{name}</h2>
            <p>{email}</p>
            </div>
        </div>
    );
}

export default Card;