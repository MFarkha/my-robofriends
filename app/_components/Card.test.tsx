import Card from "./Card";
import { expect, test } from "vitest";
import { render, screen } from "@testing-library/react";

test("expects to render the Card component", () => {
  const mockRobot = {
    id: 1,
    name: "John Snow",
    email: "john@example.com",
  };
  const { container } = render(<Card key={mockRobot.id} robot={mockRobot} />);
  expect(
    screen.getByRole("heading", { level: 2, name: mockRobot.name }),
  ).toBeDefined();
  expect(container).toMatchSnapshot();
});
