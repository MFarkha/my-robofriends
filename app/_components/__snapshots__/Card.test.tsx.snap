// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`expects to render the Card component 1`] = `
<div>
  <div
    class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
  >
    <img
      alt="robots"
      height="200"
      src="https://robohash.org/1.png?size=200x200"
      width="200"
    />
    <div>
      <h2
        class="f5"
      >
        John Snow
      </h2>
      <p>
        john@example.com
      </p>
    </div>
  </div>
</div>
`;
